libsolidity/Types.cpp

```diff
@@ -1103,11 +1103,16 @@ TypePointer FunctionType::externalType() const
 	TypePointers paramTypes;
 	TypePointers retParamTypes;
 
-	for (auto it = m_parameterTypes.cbegin(); it != m_parameterTypes.cend(); ++it)
-		paramTypes.push_back((*it)->externalType());
-	for (auto it = m_returnParameterTypes.cbegin(); it != m_returnParameterTypes.cend(); ++it)
-		retParamTypes.push_back((*it)->externalType());
-
+	for(auto type: m_parameterTypes)
+	{
+		solAssert(!!type->externalType(), "To be included in external type of the function, the argument should have external type.");
```

Comment:
Is it really an assertion? If you throw an exception here, the code above will not be reached.